### ISSUE TITLE
404 page

### ISSUE THREAD

**Issue by spences10** [chingu-voyage3/grad.then#72]
Detail and information to replace the standard Gatsby 404 page

**Comment by mar-bi** [comment]
As per our talk in slack with @miljan-fsd, i decided to add here my steps on 
*How to a create custom 404 page for the app*

**Readings:**
1. How to create **404** page - https://www.gatsbyjs.org/docs/add-404-page/ 
2. How to modify a page layout (remove `header`, `footer`, etc) - https://www.gatsbyjs.org/docs/creating-and-modifying-pages/#choosing-the-page-layout

You need:
- `404.js`
![image](https://user-images.githubusercontent.com/25195616/34340702-ae37b79e-e991-11e7-9a32-1136ca9429ca.png)
-  another layout (it must be created before the next step - here `notFound.js`) 
![image](https://user-images.githubusercontent.com/25195616/34340722-06eab904-e992-11e7-90cc-0f88c1a96790.png)
- add lines of code to `gatsby-node.js` as shown in [2](https://www.gatsbyjs.org/docs/creating-and-modifying-pages/#choosing-the-page-layout) with correct filenames and paths
![image](https://user-images.githubusercontent.com/25195616/34340740-5e75463a-e992-11e7-8a6d-cad5c26cc8cf.png)

in `gatsby-node.js`
```js
const path = require("path");

exports.onCreatePage = ({ page, boundActionCreators }) => { // i didn't use async function, it throws error
 const { createPage } = boundActionCreators;

 return new Promise((resolve, reject) => {
   if (page.path.match(/^\/404/)) {
     // It's assumed that `landingPage.js` exists in the `/layouts/` directory
     page.layout = "notFound";

     // Update the page.
     createPage(page);
   }

   resolve();
 });
};
```

and `notFound.js`
```jsx
import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import styled, { ThemeProvider } from 'styled-components'

import { theme } from '../theme/globalStyle'

const PageContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-template-rows: auto;
`

// min-height is just for the purpose of nice looking page
const Main = styled.div`
  min-height: 800px;
`

const TemplateWrapper = ({ children }) => (
  <ThemeProvider theme={theme}>
    <PageContainer>
      <Helmet
        title="grad.then()"
        meta={[
          { name: 'description', content: 'error page' },
          { name: 'keywords', content: 'error, 404, path not found' }
        ]}
      />
      <Main>{children()}</Main>
    </PageContainer>
  </ThemeProvider>
)

TemplateWrapper.propTypes = {
  children: PropTypes.func
}

export default TemplateWrapper

```

And the **404 page** without `Header` and `Footer` 

![image](https://user-images.githubusercontent.com/25195616/34340852-44a446c8-e994-11e7-80db-af8903dfd480.png)

**Comment by mar-bi** [comment]
Everyone is free to pick up this issue and enjoy building **404**. :cowboy_hat_face: :upside_down_face:

**Comment by spences10** [comment]
I'll take this if no one else wants

**Comment by mar-bi** [comment]
closed with #146
